### PR TITLE
Add missing Ws2_32.lib when building azure core on Windows.

### PR DESCRIPTION
**Pull request by BillyONeal**
This was first discovered by trying to update curl in vcpkg: https://github.com/microsoft/vcpkg/pull/27778

Curl is no longer publishing that it depends on ws2_32 when dynamically linked. Azure Core directly uses Ws2_32 things, so it needs that declared dependency. For example:

https://github.com/Azure/azure-sdk-for-cpp/blob/90fc46693f4b90b805047f165c0b41e07e867594/sdk/core/azure-core/src/http/curl/curl.cpp#L165

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes      FYI you have some trailing whitespace in this file :(
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [ ] Related issue listed
- [x] Comments in source
- [x] No typos
- [ ] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by LarryOsterman — APPROVED** [review]

**Comment by BillyONeal** [comment]
Probably caused by https://github.com/curl/curl/pull/9125

**Review by antkmsft — APPROVED** [review]
Thank you Billy!

File: sdk/core/azure-core/CMakeLists.txt [hunk 1]
```diff
@@ -179,6 +179,9 @@ endif()
 
 if(BUILD_TRANSPORT_CURL)
   target_link_libraries(azure-core PUBLIC CURL::libcurl)
+  if(WIN32)
+    target_link_libraries(azure-core PRIVATE Ws2_32)
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CMakeLists.txt:183]: @BillyONeal do we need to declare this as a dependency in our vcpkg.json file, like we do `wil`?
https://github.com/Azure/azure-sdk-for-cpp/blob/f587bf9e797c8878436a8cacba3781e99eb2ae31/vcpkg.json#L22-L23
And add a `find_dependency` call here:
https://github.com/Azure/azure-sdk-for-cpp/blob/53c9da155244b72ac90ad27f515d4ea1e7505a01/sdk/core/azure-core/vcpkg/Config.cmake.in#L14

Comment (LarryOsterman) [reply sdk/core/azure-core/CMakeLists.txt:183]: @ahsonkhan QQ: What is the vcpkg package containing the WIn32 winsock headers?

As far as I know, there isn't one - that lib comes from the windows SDK. And that's not a vcpkg package. 

wil is a vcpkg package so we can (and must) take a dependency on it.

Comment (BillyONeal) [reply sdk/core/azure-core/CMakeLists.txt:183]: What @LarryOsterman said. That lib comes from the Windows SDK. find_dependency goes with find_package which isn't used here.

**Pull request merged**